numpy/core/src/common/npy_cpu_features.c.src

```diff
@@ -76,12 +76,15 @@ npy__cpu_getxcr0(void)
 #if defined(_MSC_VER) || defined (__INTEL_COMPILER)
     return _xgetbv(0);
 #elif defined(__GNUC__) || defined(__clang__)
+    /* named form of xgetbv not supported on OSX, so must use byte form, see:
+     https://github.com/asmjit/asmjit/issues/78
+   */
     unsigned int eax, edx;
-    __asm__("xgetbv" : "=a" (eax), "=d" (edx) : "c" (0));
-    return (eax | (unsigned long long)edx << 32);
+    __asm(".byte 0x0F, 0x01, 0xd0" : "=a"(eax), "=d"(edx) : "c"(0));
+    return eax;
 #else
-    // TODO: handle other x86 compilers
-    return 0;
+#error "Unsupported compiler, x86 cpuid requires either GCC, Clang or MSVC."
```

Comment:
let's keep it, and fail back to the possible minimal features through `npy__cpu_init_features()`
so I suggest replacing: 

```C
static void npy__cpu_init_features(void)
{
    ....
    if (reg[0] == 0)
        return;
    ....
}
````

with

````C
static void npy__cpu_init_features(void)
{
   ....
   if (reg[0] == 0) {
       npy__cpu_have[NPY_CPU_FEATURE_MMX]  = 1;
       npy__cpu_have[NPY_CPU_FEATURE_SSE]  = 1;
       npy__cpu_have[NPY_CPU_FEATURE_SSE2] = 1;
#ifdef NPY_CPU_AMD64
       npy__cpu_have[NPY_CPU_FEATURE_SSE3] = 1;
#endif
       return
   }
    ....
}
````

Reply:
Good advise! minimal features is better than error message.